src/v/cluster/partition.cc

```diff
@@ -399,7 +400,7 @@ kafka_stages partition::replicate_in_stages(
           }
           auto old_offset = r.value().last_offset;
           auto new_offset = kafka::offset(
-            _translator->from_log_offset(old_offset)());
+            get_offset_translator_state()->from_log_offset(old_offset)());
```

Comment:
if we aren't going to cache any pointers to the state going forward, should we remove that interface and drop the first commit?

Reply:
Yeah, I remember the whole point of introducing these shared ptrs was to allow caching it for the duration of the fetch request (and the `offset_translator_state` object could come from two places - for a local read we got it from raft and for a cloud read we reconstructed it on the fly while reading the segments). And just translating offsets on the fly was not enough because just before returning data to the cluent we had to go back and translate aborted tx ranges. Probably this can all be simplified somehow :)

Reply:
Makes sense. I was confused because the wording on one of the commits sounded like all cases in which the pointer were cached were being removed.